src/Chapter1.hs

```diff
@@ -641,8 +652,11 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit n
+    | mod n 10 == n = n
+    | otherwise = firstDigit (div n 10)
```

Comment:
That is the right idea! 💡 
But, it won't work correctly on negative numbers. It would show the wrong output for ` -10 < x < 0` (in case of negatives, the result will be negative). And on the other negatives in will hang.